### PR TITLE
Decouple Database From the ServiceRegistry

### PR DESCRIPTION
**Pull request by rawfalafel**
While looking into #280, I noticed that there might not be a need for the database to be in the service registry. This would also lift the requirement that services start in a certain order (at least for beaconchain). Let me know if there's a reason for the database to be in the service registry. If not, I can update the tests and look it over a bit more for a proper review.

### CODE REVIEW

**Comment by rauljordan** [comment]
The reason here is because it starts a leveldb connection, which we want to handle through graceful exiting. It is important to consider the different scenarios under which we need to handle this appropriately.

**Comment by rawfalafel** [comment]
Gotcha. I changed the code to keep the behavior the same, so as far as I can tell there shouldn't be any regressions. A nice-to-have feature is a blocking `ServiceRegistry.StopAll()` so that the database can disconnect after all services exit.

Otherwise does this look good?

**Comment by rauljordan** [comment]
Linter fails - I think gazelle needs to run

**Comment by rawfalafel** [comment]
ran linter and fixed tests :+1:

**Comment by rauljordan** [comment]
@rawfalafel if we're doing this, then let's just move it to the shared folder

**Comment by rauljordan** [comment]
Hey @rawfalafel let's move this to shared and fix some conflicts before merging

**Comment by rawfalafel** [comment]
ready for another look :+1:

File: shared/database/database.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		log.Error(fmt.Sprintf("Could not start beaconDB: %v", err))
+		return nil, err
+	}
+	db._db = _db
+
+	return db, nil
+}
+
+// Close closes the database
+func (b *DB) Close() {
```
Comment (rawfalafel) [inline shared/database/database.go]: Note: To differentiate from a service, the cleanup function's now called `Close`

**Pull request merged**